### DISCUSSION TITLE
Is it possible to access the pathname in a Next.js server component?

### DISCUSSION THREAD

**Discussion by Sathyaraj-dev** [vercel/next.js#71720] [App Router] [answered]
I'm working with server components in Next.js and would like to know if there's a way to access the pathname directly within these components. In client components, I can use the useRouter hook to get the pathname, but I'm unsure how to achieve this in server components.

Are there any built-in methods or approaches for accessing the current pathname in server components, particularly in the context of dynamic routes? Any guidance or examples would be greatly appreciated!

**Comment by himanshu634** [comment, marked as answer]
You can do it by setting cookies from middleware, that is one option.

**Reply by Sathyaraj-dev** [reply]
Himanshu,

Thanks for the answer! Besides using cookies in middleware, are there any other options you would suggest?

**Reply by nirvikpurkait** [reply]
If you want to access the value of the pathname, besides using cookies you can use the `params` from a `page.tsx` or `layout.tsx`. Learn [more](https://nextjs-faq.com/get-pathname-in-rsc)

**Reply by Sathyaraj-dev** [reply]
Thank you for the solution @nirvikpurkait
